tests/ui/escape_analysis.rs

```diff
@@ -174,3 +174,8 @@ mod issue_3739 {
         };
     }
 }
+
+/// Issue #5542
+///
+/// This shouldn't warn for `boxed_local` as it is a function implemented in C.
```

Comment:
```suggestion
/// This shouldn't warn for `boxed_local` as it is intended to be called from non-Rust code.
```
If I'm not mistaken `extern fn`s are defined in Rust and intended to be called from non-Rust code, and `fn`s in `extern` blocks are declarations of non-Rust functions intended to be called from Rust.

Reply:
Oh wow you're right, I flipped the definitions when writing/reading the examples. Should be fixed now.